[BUG] MKS SGEN L V1.0 LPC1768 MKS TS35 V2.0  - LCD_BED_LEVELING and TFT_COLOR_UI compilation error

### Hardware:
Motherbord - MKS SGEN L V1.0 LPC1768
LCD - MKS TS35 V2.0
Latest Marlin 2.0 Bugfix

### Bug Description
I user **MKS TS35 V2.0** and **TFT_COLOR_UI** mode
Firmware don't want compile with **#define LCD_BED_LEVELING** and **#define TFT_COLOR_UI**
If use define #TFT_CLASSIC_UI and #define LCD_BED_LEVELING then all ok


### Configuration Files
[My configuration.zip](https://drive.google.com/file/d/163QsFE1xwZHnp7JinYYcTxPEytxZtL5d/view?usp=sharing)


### Log error
Compiling .pio\build\LPC1768\src\src\gcode\temp\M303.cpp.o
Compiling .pio\build\LPC1768\src\src\gcode\units\M82_M83.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\extui\ui_api.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\fontutils.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\lcdprint.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\marlinui.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu_advanced.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu_bed_corners.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu_bed_leveling.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu_configuration.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu_custom.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu_filament.cpp.o
Compiling .pio\build\LPC1768\src\src\lcd\menu\menu_game.cpp.o
In file included from Marlin\src\lcd\menu\../../inc/../core/boards.h:24,
                 from Marlin\src\lcd\menu\../../inc/MarlinConfigPre.h:37,
                 from Marlin\src\lcd\menu\menu_bed_leveling.cpp:27:
Marlin\src\lcd\menu\menu_bed_leveling.cpp: In function 'void _lcd_level_bed_homing_done()':
**Marlin\src\lcd\menu\menu_bed_leveling.cpp:172:32: error: 'touch' was not declared in this scope**
  172 |       TERN_(HAS_GRAPHICAL_TFT, touch.add_control(CLICK, 0, 0, TFT_WIDTH, TFT_HEIGHT));
      |                                ^~~~~
Marlin\src\lcd\menu\../../inc/../core/macros.h:419:26: note: in definition of macro 'THIRD'
  419 | #define THIRD(a,b,c,...) c
      |                          ^
Marlin\src\lcd\menu\../../inc/../core/macros.h:196:29: note: in expansion of macro '___TERN'
  196 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin\src\lcd\menu\../../inc/../core/macros.h:195:29: note: in expansion of macro '__TERN'
  195 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin\src\lcd\menu\../../inc/../core/macros.h:194:29: note: in expansion of macro '_TERN'
  194 | #define TERN_(O,A)          _TERN(_ENA_1(O),,A)     // OPTION converted to A or '<nul>'
      |                             ^~~~~
Marlin\src\lcd\menu\menu_bed_leveling.cpp:172:7: note: in expansion of macro 'TERN_'
  172 |       TERN_(HAS_GRAPHICAL_TFT, touch.add_control(CLICK, 0, 0, TFT_WIDTH, TFT_HEIGHT));
      |       ^~~~~
**Marlin\src\lcd\menu\menu_bed_leveling.cpp:172:50: error: 'CLICK' was not declared in this scope**
  172 |       TERN_(HAS_GRAPHICAL_TFT, touch.add_control(CLICK, 0, 0, TFT_WIDTH, TFT_HEIGHT));
      |                                                  ^~~~~
Marlin\src\lcd\menu\../../inc/../core/macros.h:419:26: note: in definition of macro 'THIRD'
  419 | #define THIRD(a,b,c,...) c
      |                          ^
Marlin\src\lcd\menu\../../inc/../core/macros.h:196:29: note: in expansion of macro '___TERN'
  196 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin\src\lcd\menu\../../inc/../core/macros.h:195:29: note: in expansion of macro '__TERN'
  195 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin\src\lcd\menu\../../inc/../core/macros.h:194:29: note: in expansion of macro '_TERN'
  194 | #define TERN_(O,A)          _TERN(_ENA_1(O),,A)     // OPTION converted to A or '<nul>'
      |                             ^~~~~
Marlin\src\lcd\menu\menu_bed_leveling.cpp:172:7: note: in expansion of macro 'TERN_'
  172 |       TERN_(HAS_GRAPHICAL_TFT, touch.add_control(CLICK, 0, 0, TFT_WIDTH, TFT_HEIGHT));
      |       ^~~~~
*** [.pio\build\LPC1768\src\src\lcd\menu\menu_bed_leveling.cpp.o] Error 1
=========================================================================== [FAILED] Took 148.90 seconds ===========================================================================

Environment    Status    Duration
-------------  --------  ------------
LPC1768        FAILED    00:02:28.897

--- a/Marlin/src/lcd/menu/menu_bed_leveling.cpp
+++ b/Marlin/src/lcd/menu/menu_bed_leveling.cpp
@@ -169,7 +169,9 @@
     if (ui.should_draw()) {
       MenuItem_static::draw(1, GET_TEXT(MSG_LEVEL_BED_WAITING));
       // Color UI needs a control to detect a touch
-      TERN_(HAS_GRAPHICAL_TFT, touch.add_control(CLICK, 0, 0, TFT_WIDTH, TFT_HEIGHT));
+      #if BOTH(TOUCH_SCREEN, HAS_GRAPHICAL_TFT)
+        touch.add_control(CLICK, 0, 0, TFT_WIDTH, TFT_HEIGHT);
+      #endif
     }
     if (ui.use_click()) {
       manual_probe_index = 0;

--- a/Marlin/src/lcd/tft/ui_480x320.cpp
+++ b/Marlin/src/lcd/tft/ui_480x320.cpp
@@ -49,7 +49,9 @@
   #error "Seriously? High resolution TFT screen without menu?"
 #endif
 
-static bool draw_menu_navigation = false;
+#if ENABLED(TOUCH_SCREEN)
+  static bool draw_menu_navigation = false;
+#endif
 
 void MarlinUI::tft_idle() {
   #if ENABLED(TOUCH_SCREEN)
@@ -897,35 +899,37 @@ static void z_minus() {
   moveAxis(Z_AXIS, -1);
 }
 
-static void e_select() {
-  motionAxisState.e_selection++;
-  if (motionAxisState.e_selection >= EXTRUDERS) {
-    motionAxisState.e_selection = 0;
+#if ENABLED(TOUCH_SCREEN)
+  static void e_select() {
+    motionAxisState.e_selection++;
+    if (motionAxisState.e_selection >= EXTRUDERS) {
+      motionAxisState.e_selection = 0;
+    }
+
+    quick_feedback();
+    drawCurESelection();
+    drawAxisValue(E_AXIS);
   }
 
-  quick_feedback();
-  drawCurESelection();
-  drawAxisValue(E_AXIS);
-}
+  static void do_home() {
+    quick_feedback();
+    drawMessage(GET_TEXT(MSG_LEVEL_BED_HOMING));
+    queue.inject_P(G28_STR);
+    // Disable touch until home is done
+    TERN_(HAS_TFT_XPT2046, touch.disable());
+    drawAxisValue(E_AXIS);
+    drawAxisValue(X_AXIS);
+    drawAxisValue(Y_AXIS);
+    drawAxisValue(Z_AXIS);
+  }
 
-static void do_home() {
-  quick_feedback();
-  drawMessage(GET_TEXT(MSG_LEVEL_BED_HOMING));
-  queue.inject_P(G28_STR);
-  // Disable touch until home is done
-  TERN_(HAS_TFT_XPT2046, touch.disable());
-  drawAxisValue(E_AXIS);
-  drawAxisValue(X_AXIS);
-  drawAxisValue(Y_AXIS);
-  drawAxisValue(Z_AXIS);
-}
-
-static void step_size() {
-  motionAxisState.currentStepSize = motionAxisState.currentStepSize / 10.0;
-  if (motionAxisState.currentStepSize < 0.0015) motionAxisState.currentStepSize = 10.0;
-  quick_feedback();
-  drawCurStepValue();
-}
+  static void step_size() {
+    motionAxisState.currentStepSize = motionAxisState.currentStepSize / 10.0;
+    if (motionAxisState.currentStepSize < 0.0015) motionAxisState.currentStepSize = 10.0;
+    quick_feedback();
+    drawCurStepValue();
+  }
+#endif
 
 #if HAS_BED_PROBE
   static void z_select() {
@@ -1021,7 +1025,7 @@ void MarlinUI::move_axis_screen() {
   motionAxisState.zTypePos.x = x;
   motionAxisState.zTypePos.y = y;
   drawCurZSelection();
-  #if HAS_BED_PROBE
+  #if BOTH(HAS_BED_PROBE, TOUCH_SCREEN)
     if (!busy) touch.add_control(BUTTON, x, y, BTN_WIDTH, 34 * 2, (intptr_t)z_select);
   #endif
 

--- a/buildroot/tests/mks_robin_nano35-tests
+++ b/buildroot/tests/mks_robin_nano35-tests
@@ -65,5 +65,14 @@ opt_set X_DRIVER_TYPE TMC2209
 opt_set Y_DRIVER_TYPE TMC2209
 exec_test $1 $2 "MKS Robin v2 nano LVGL SPI + TMC"
 
+#
+# MKS Robin v2 nano New Color UI 480x320 SPI Without Touch Screen
+#
+use_example_configs Mks/Robin
+opt_set MOTHERBOARD BOARD_MKS_ROBIN_NANO_V2
+opt_disable TFT_INTERFACE_FSMC TFT_RES_320x240 TOUCH_SCREEN
+opt_enable TFT_INTERFACE_SPI TFT_RES_480x320 TFT_COLOR_UI
+exec_test $1 $2 "MKS Robin v2 nano New Color UI 480x320 SPI without TOUCH_SCREEN"
+
 # cleanup
 restore_configs

